File: mex_examples_multem/example_vz.m
% Copyright 2021 Ivan Lobato <Ivanlh20@gmail.com>

clear; clc;
addpath([fileparts(pwd) filesep 'mex_bin'])
addpath([fileparts(pwd) filesep 'crystalline_materials'])
addpath([fileparts(pwd) filesep 'matlab_functions'])

Z = 51;
charge = 1;

rmin = 1e-02; rmax = 5.0; nr = 512;
dlnr = log(rmax/rmin)/(nr-1); r = rmin*exp((0:1:(nr-1))*dlnr);
z0 = -8.0; ze = 8.0;

tic;
[f1, df1] = ilc_vz(1, Z, charge, z0, ze, r);
[f2, df2] = ilc_vz(2, Z, charge, z0, ze, r);
[f3, df3] = ilc_vz(3, Z, charge, z0, ze, r);
[f4, df4] = ilc_vz(4, Z, charge, z0, ze, r);
[f5, df5] = ilc_vz(5, Z, charge, z0, ze, r);
[f6, df6] = ilc_vz(6, Z, charge, z0, ze, r);
toc;

figure(1); clf;

subplot(1, 2, 1);
hold on;
plot(r, f1, '-k', r, f2, '-b', r, f3, '-c', r, f4, '-m', r, f5, '-r', r, f6, '-g');
set(gca,'FontSize',12,'LineWidth',1,'PlotBoxAspectRatio',[1.25 1 1]);
title('Atomic potential');
ylabel('$\displaystyle V(r)$','interpreter','latex','FontSize',14);
xlabel('$\mathbf{r}$','interpreter','latex','FontSize',12);
xlim([0 rmax]);
legend('Doyle [0-4]', 'Peng [0-4]', 'Peng [0-12]', 'Kirkland [0-12]', 'Weickenmeier [0-12]', 'Lobato [0-12]');

subplot(1, 2, 2);
hold on;
plot(r, df1, '-k', r, df2, '-b', r, df3, '-c', r, df4, '-m', r, df5, '-r', r, df6, '-g');
set(gca,'FontSize',12,'LineWidth',1,'PlotBoxAspectRatio',[1.25 1 1]);
title('Derivative of the Atomic potential');
ylabel('$\displaystyle \frac{d V(r)}{dr}$','interpreter','latex','FontSize',14);
xlabel('$\mathbf{r}$','interpreter','latex','FontSize',12);
xlim([0 rmax]);
legend('Doyle [0-4]', 'Peng [0-4]', 'Peng [0-12]', 'Kirkland [0-12]', 'Weickenmeier [0-12]', 'Lobato [0-12]');

set(gcf,'units','normalized','outerposition',[0 0 1 1]);

File: multem-python/test/examples/example_vz.py
import numpy as np
import time
import multem
import multem.crystalline_materials

from math import log


def run():

    Z = 79
    charge = 1

    rmin = 1e-02
    rmax = 5.0
    nr = 512
    dlnr = log(rmax / rmin) / (nr - 1)
    r = rmin * np.exp(np.arange(0, (nr - 1), 1) * dlnr)
    z0 = -8.0
    ze = 8.0

    st = time.perf_counter()
    [f1, df1] = multem.vz(1, Z, charge, z0, ze, r)
    [f2, df2] = multem.vz(2, Z, charge, z0, ze, r)
    [f3, df3] = multem.vz(3, Z, charge, z0, ze, r)
    [f4, df4] = multem.vz(4, Z, charge, z0, ze, r)
    [f5, df5] = multem.vz(5, Z, charge, z0, ze, r)
    [f6, df6] = multem.vz(6, Z, charge, z0, ze, r)
    print("Time: %.4f seconds" % (time.perf_counter() - st))


if __name__ == "__main__":
    run()
